src/hooks/core/chat/useDMMessage.ts

```diff
@@ -0,0 +1,144 @@
+import {v4 as uuid} from 'uuid';
+import useLocalDatabaseHook from '../../../database/hooks/useLocalDatabaseHook';
+import ChannelListMemberSchema from '../../../database/schema/ChannelListMemberSchema';
+import ChannelList from '../../../database/schema/ChannelListSchema';
+import ChatSchema from '../../../database/schema/ChatSchema';
+import UserSchema from '../../../database/schema/UserSchema';
+import {initChatFromPost, initChatFromPostAnon} from '../../../service/chat';
+import {getChannelListInfo} from '../../../utils/string/StringUtils';
+import useUserAuthHook from '../auth/useUserAuthHook';
+import useChatUtilsHook from './useChatUtilsHook';
+
+type ChannelCategory = 'SIGNED' | 'ANONYMOUS';
+
+const useDMMessage = () => {
+  const {localDb, refresh} = useLocalDatabaseHook();
+  const {signedProfileId, anonProfileId} = useUserAuthHook();
+  const {goToChatScreen} = useChatUtilsHook();
+
+  const saveChannelList = async (
+    channel,
+    type: ChannelCategory,
+    builtChannelData: {
+      better_channel_member: any;
+      members: any;
+    }
+  ) => {
+    try {
+      if (channel?.members?.length === 0) return Promise.reject(Error('no members'));
+      let userTarget = channel?.better_channel_members.find(
+        (data) => data?.user?.id !== signedProfileId
+      );
+
+      if (type === 'ANONYMOUS') {
+        userTarget = channel?.better_channel_members.find(
+          (data) => data?.user?.id !== anonProfileId
+        );
+      }
+
+      const {
+        anonUserInfoEmojiName,
+        anonUserInfoColorCode,
+        anonUserInfoColorName,
+        anonUserInfoEmojiCode,
+        channelName,
+        channelImage
+      } = getChannelListInfo(builtChannelData, signedProfileId, anonProfileId);
+
+      const chatType = type === 'ANONYMOUS' ? 'ANON_PM' : 'PM';
+      const chatName = {
+        name: channelName,
+        image: channelImage
+      };
+
+      channel.firstMessage = channel?.messages?.[channel?.messages?.length - 1];
+      channel.myUserId = signedProfileId; // change to use getChannelListInfo
+      channel.targetName = chatName?.name; // change to use getChannelListInfo
+      channel.targetImage = chatName?.image; // change to use getChannelListInfo
+
+      channel.channel = {...channel};
+      console.log('data', {
+        anon_user_info_color_name: anonUserInfoColorName,
+        anon_user_info_color_code: anonUserInfoColorCode,
+        anon_user_info_emoji_code: anonUserInfoEmojiCode,
+        anon_user_info_emoji_name: anonUserInfoEmojiName
+      });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

There's a console log statement that seems to be used for debugging purposes. It's generally a good idea to remove such statements before committing code to avoid cluttering the console output in a production environment.

```diff
-      console.log('data', {
-        anon_user_info_color_name: anonUserInfoColorName,
-        anon_user_info_color_code: anonUserInfoColorCode,
-        anon_user_info_emoji_code: anonUserInfoEmojiCode,
-        anon_user_info_emoji_name: anonUserInfoEmojiName
-      });
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->